core/domain/user_jobs_one_off_test.py

```diff
@@ -196,6 +196,63 @@ def test_user_contributions_get_created_after_running_the_job(self):
             ['exp_id'])
 
 
+class UserAuthModelOneOffJobTests(test_utils.GenericTestBase):
+    """Tests for the one-off UserAuthModel migration job."""
+
+    USER_EMAIL = 'a@example.com'
+    USER_USERNAME = 'a'
+    NEW_USERNAME = 'new_username'
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = user_jobs_one_off.UserAuthModelOneOffJob.create_new()
+        user_jobs_one_off.UserAuthModelOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+    def setUp(self):
+        super(UserAuthModelOneOffJobTests, self).setUp()
+        self.signup(self.USER_EMAIL, self.USER_USERNAME)
+        self.user_a_id = self.get_user_id_from_email(self.USER_EMAIL)
+
+    def test_one_off_job_new_gae_id_creates_new_auth_model_entry(self):
+        user_settings_model = user_models.UserSettingsModel.get_by_id(
+            self.user_a_id)
+        new_gae_id = user_settings_model.gae_id
+        self.assertIsNone(user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, new_gae_id))
+
+        self._run_one_off_job()
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, new_gae_id)
+        self.assertEqual(user_auth_model.gae_id[0], new_gae_id)
+        self.assertEqual(user_auth_model.id, user_settings_model.id)
+
+    def test_existing_gae_id_mapped_to_different_user_id_is_updated(self):
```

Comment:
Reiterating above--I'd expect this case to fail, not remap.

Reply:
Fixed it in the code. Also the test `test_one_off_job_gae_id_registered_for_another_user_raises_error` now raises an error for this condition. PTAL!